src/database/database.ts

```diff
@@ -1,10 +1,32 @@
+import {DatabaseEmulatorCredential} from '../auth/credential';
 import {FirebaseApp} from '../firebase-app';
 import {FirebaseDatabaseError} from '../utils/error';
 import {FirebaseServiceInterface, FirebaseServiceInternalsInterface} from '../firebase-service';
 import {Database} from '@firebase/database';
 
 import * as validator from '../utils/validator';
 
+const APP_OPTIONS_PROJECT_ID = 'projectId';
+const APP_OPTIONS_CREDENTIAL = 'credential';
+
+/**
+ * Constant holding the fully-qualified domain URI for a database emulator
+ * instance. If specified, the contents of this variable will be used to
+ * set `databaseURL` in FirebaseAppOptions. The varaible should be a complete
+ * URI specifying a transfer protocol, hostname, and port number:
+ *
+ * FIREBASE_DATABASE_EMULATOR_HOST=http://localhost:9000
```

Comment:
If I am not mistaken, the protocol is omitted in FIRESTORE_EMULATOR_HOST. I see that it might make sense to include it here (since we need to include the namespace). Should we come up with a consistent pattern that apploes to both RTDB and Firestore? @ryanpbrewster

Reply:
the protocol should not be included in the `FIRESOTRE_EMULATOR_HOST` variable.

Reply:
Do we care about the consistency between FIRESTORE_EMULATOR_HOST and FIREBASE_DATABASE_EMULATOR_HOST?

Reply:
Yes, ideally neither should include the protocol.

Reply:
This comment is dated, neither emulator host variable has a protocol.